geomesa-core/src/main/scala/org/locationtech/geomesa/core/index/QueryPlanners.scala

```diff
@@ -285,18 +298,18 @@ trait GeoHashPlanner extends Logging {
   def polyToPlan(geom: Geometry, offset: Int, bits: Int): KeyPlan = {
     val subHashes = geomToGeoHashes(geom, offset, bits).sorted
     logger.trace(s"Geom to GeoHashes has returned: ${subHashes.size} subhashes to cover $geom $offset $bits.")
-    subHashes match {
-      case subs if subs.size == 0 =>
-        // if the list is empty, then there are probably too many 35-bit GeoHashes
-        // that fall inside the given polygon; in this case, return the LL, UR
-        // GeoHash endpoints of the entire range (which could encompass many
-        // more GeoHashes than we wish, but can only be better than (or equal
-        // to) a full-table scan)
-        val env = geom.getEnvelopeInternal
-        val ghLL = GeoHash(env.getMinX, env.getMinY)
-        val ghUR = GeoHash(env.getMaxX, env.getMaxY)
-        KeyRange(ghLL.hash, ghUR.hash)
-      case subs => KeyList(subs.sorted)
+    if (subHashes.isEmpty) {
+      // if the list is empty, then there are probably too many 35-bit GeoHashes
+      // that fall inside the given polygon; in this case, return the LL, UR
+      // GeoHash endpoints of the entire range (which could encompass many
+      // more GeoHashes than we wish, but can only be better than (or equal
+      // to) a full-table scan)
+      val env = geom.getEnvelopeInternal
+      val ghLL = GeoHash(env.getMinX, env.getMinY)
+      val ghUR = GeoHash(env.getMaxX, env.getMaxY)
+      KeyRange(ghLL.hash, ghUR.hash)
+    } else {
+      KeyList(subHashes)
```

Comment:
It was being sorted twice. Note that subHashes is already sorted when it's created on line 299.